x-pack/plugins/reporting/public/panel_actions/get_csv_panel_action.tsx

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import dateMath from '@elastic/datemath';
+import { i18n } from '@kbn/i18n';
+
+import { ContextMenuAction, ContextMenuActionsRegistryProvider } from 'ui/embeddable';
+import { PanelActionAPI } from 'ui/embeddable/context_menu_actions/types';
+import { kfetch } from 'ui/kfetch';
+import { toastNotifications } from 'ui/notify';
+
+const API_BASE_URL = '/api/reporting/v1/generate/immediate/csv/saved-object/';
```

Comment:
I think it'd be worthwhile to import the base URL path (`API_BASE_URL_V1`) from `'../../common/constants'`